Another panic in the distributor

```
$ kubectl logs --namespace=cortex distributor-3751710785-oy58f -p
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x2f8 pc=0x81f572]

goroutine 87 [running]:
panic(0xaf75a0, 0xc420016050)
	/usr/local/go/src/runtime/panic.go:500 +0x1a1
github.com/weaveworks/cortex/vendor/github.com/aws/aws-sdk-go/aws/request.(*Request).Send(0x0, 0x464163, 0x58751f21)
	/go/src/github.com/weaveworks/cortex/vendor/github.com/aws/aws-sdk-go/aws/request/request.go:236 +0x52
github.com/weaveworks/cortex/chunk.dynamoRequestAdapter.Send(0x0, 0x58751f21, 0x556a3dd)
	<autogenerated>:137 +0x41
github.com/weaveworks/cortex/chunk.(*dynamoQueryPagesOp).do.func1(0x7fc19fe61958, 0xc42158c0c0, 0x11b5d00, 0xc42012a0d0)
	/go/src/github.com/weaveworks/cortex/chunk/chunk_store.go:780 +0x32
github.com/weaveworks/cortex/vendor/github.com/weaveworks/scope/common/instrument.TimeRequestHistogramStatus(0x7fc19fe61c90, 0xc420070cc0, 0xbcb790, 0x13, 0xc42016a230, 0xc1fb68, 0xc42053bef8, 0xc420275500, 0x0)
	/go/src/github.com/weaveworks/cortex/vendor/github.com/weaveworks/scope/common/instrument/instrument.go:45 +0x14d
github.com/weaveworks/cortex/vendor/github.com/weaveworks/scope/common/instrument.TimeRequestHistogram(0x7fc19fe61c90, 0xc420070cc0, 0xbcb790, 0x13, 0xc42016a230, 0xc42053bef8, 0x7fc19c59a150, 0xc4208262d0)
	/go/src/github.com/weaveworks/cortex/vendor/github.com/weaveworks/scope/common/instrument/instrument.go:27 +0x70
github.com/weaveworks/cortex/chunk.(*dynamoQueryPagesOp).do(0xc4202972c0)
	/go/src/github.com/weaveworks/cortex/chunk/chunk_store.go:781 +0xe1
github.com/weaveworks/cortex/chunk.(*AWSStore).dynamoRequestLoop(0xc4201dc8c0)
	/go/src/github.com/weaveworks/cortex/chunk/chunk_store.go:742 +0xaf
created by github.com/weaveworks/cortex/chunk.NewAWSStore
	/go/src/github.com/weaveworks/cortex/chunk/chunk_store.go:181 +0x155
```

--- a/chunk/dynamodb_client.go
+++ b/chunk/dynamodb_client.go
@@ -68,7 +68,10 @@ func (d dynamoRequestAdapter) OperationName() string {
 }
 
 func (d dynamoRequestAdapter) NextPage() dynamoRequest {
-	return dynamoRequestAdapter{d.Request.NextPage()}
+	if r := d.Request.NextPage(); r != nil {
+		return dynamoRequestAdapter{r}
+	}
+	return nil
 }
 
 func (d dynamoRequestAdapter) Error() error {

